IPython/kernel/zmq/kernelbase.py

```diff
@@ -347,41 +347,43 @@ def execute_request(self, stream, ident, parent):
             self.log.error("Got bad msg: ")
             self.log.error("%s", parent)
             return
-        
-        stop_on_error = content.get('stop_on_error', True)
 
         md = self._make_metadata(parent['metadata'])
-        
+
         # Re-broadcast our input for the benefit of listening clients, and
         # start computing output
         if not silent:
             self.execution_count += 1
             self._publish_execute_input(code, parent, self.execution_count)
-        
+
         reply_content = self.do_execute(code, silent, store_history,
                                         user_expressions, allow_stdin)
+        self.send_execute_reply(stream, ident, parent, md, reply_content)
+
+    def send_execute_reply(self, stream, ident, parent, md, reply_content):
+        """ Send a reply to execute_request """
```

Comment:
I have only a vague idea about what stream, ident and parent parameters mean :) They are the same as in `execute_request` which isn't documented.

Maybe it is a sign that a better extension point is needed for the public API, the one which doesn't expose these arguments.